File: Programmers/난이도모음/LEVEL 2/H-index.py
def solution(citations):
    citations.sort()
    l = len(citations)

    for i in range(l):
        h = l - i               # 1. h편 이상 인용했는가? 
        if citations[i] >= h:   # 2. h번 이상 인용된 논문인가?
            return h

    return 0

File: Programmers/난이도모음/LEVEL 2/H-index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        // 1. 정렬
        Arrays.sort(citations);

        // 2. H-index 찾기
        for (int i = 0; i < citations.length; ++i) {
            int h = citations.length - i;   // 2-(1). h편 이상 인용
            if (citations[i] >= h)          // 2-(2). h번 이상 인용된 논문인가?
                return h;
        }
        return 0;
    }
}